tensorflow_model_optimization/python/core/sparsity/keras/pruning_callbacks_test.py

```diff
@@ -55,6 +55,45 @@ def _pruned_model_setup(self, custom_training_loop=False):
       pruned_model.compile(loss=loss, optimizer=optimizer, metrics=['accuracy'])
       return pruned_model, x_train, y_train
 
+  @keras_parameterized.run_all_keras_modes
+  def testDeepLayerUpdatePruningSteps(self):
+    class TestLayer(tf.keras.layers.Layer):
+      def __init__(self):
+        super().__init__()
+        self.dense = prune.prune_low_magnitude(keras.layers.Dense(10))
+        self.head = keras.layers.Dense(5, activation='softmax')
+
+      def call(self, x):
+        x = self.dense(x)
+        return self.head(x)
+
+    class TestModel(tf.keras.Model):
+      def __init__(self):
+        super().__init__()
+        self.test_layer = TestLayer()
+
+      def call(self, x):
+        return self.test_layer(x)
+
+    x_train = np.random.rand(self._BATCH_SIZE, 10)
```

Comment:
To make this unit test focus on the important aspects of what's being tested (the specific model and callback), could you try to reuse self._pruned_model_setup like the other unit tests? It hides away the compile/loss/optimizer, which don't matter too much.

You can move out the line "pruned_model = prune.prune_low_magnitude(
        keras_test_utils.build_simple_dense_model())" from `_pruned_model_setup` to help.